test/unit/org/apache/cassandra/index/sai/memory/TrieMemtableIndexTest.java

```diff
@@ -18,158 +18,55 @@
-
     @BeforeClass
     public static void setShardCount()
     {
         System.setProperty("cassandra.trie.memtable.shard.count", "8");
     }
 
-    @Before
-    public void setup() throws Throwable
-    {
-        assertEquals(8, TrieMemtable.SHARD_COUNT);
-
-        TokenMetadata metadata = StorageService.instance.getTokenMetadata();
-        metadata.updateNormalTokens(BootStrapper.getRandomTokens(metadata, 10), FBUtilities.getBroadcastAddressAndPort());
-
-        TableMetadata tableMetadata = TableMetadata.builder("ks", "tb")
-                                                   .addPartitionKeyColumn("pk", Int32Type.instance)
-                                                   .addRegularColumn("val", Int32Type.instance)
-                                                   .build();
-        cfs = MockSchema.newCFS(tableMetadata);
-        partitioner = cfs.getPartitioner();
-        memtable = (AbstractAllocatorMemtable) cfs.getCurrentMemtable();
-        indexContext = SAITester.createIndexContext("index", Int32Type.instance, cfs);
-        indexSearchCounter.reset();
-        keyMap = new TreeMap<>();
-        rowMap = new HashMap<>();
-
-        Injections.inject(indexSearchCounter);
-    }
-
-    @After
-    public void resetBufferType() throws Exception
-    {
-        setTrieMemtableBufferType(BufferType.OFF_HEAP);
-    }
-
-    @Test
-    public void onHeapAllocation() throws Exception
-    {
-        setTrieMemtableBufferType(BufferType.ON_HEAP);
-        memtableIndex = new TrieMemtableIndex(indexContext, memtable);
-        assertEquals(TrieMemtable.SHARD_COUNT, memtableIndex.shardCount());
-
-        assertTrue(memtable.getAllocator().onHeap().owns() == 0);
-        assertTrue(memtable.getAllocator().offHeap().owns() == 0);
-
-        for (int row = 0; row < 100; row++)
-        {
-            addRow(row, row);
-        }
-
-        assertTrue(memtable.getAllocator().onHeap().owns() > 0);
-        assertTrue(memtable.getAllocator().offHeap().owns() == 0);
-    }
-
-    @Test
-    public void offHeapAllocation() throws Exception
-    {
-        setTrieMemtableBufferType(BufferType.OFF_HEAP);
-        memtableIndex = new TrieMemtableIndex(indexContext, memtable);
-        assertEquals(TrieMemtable.SHARD_COUNT, memtableIndex.shardCount());
-
-        assertTrue(memtable.getAllocator().onHeap().owns() == 0);
-        assertTrue(memtable.getAllocator().offHeap().owns() == 0);
-
-        for (int row = 0; row < 100; row++)
-        {
-            addRow(row, row);
-        }
-
-        assertTrue(memtable.getAllocator().onHeap().owns() > 0);
-        assertTrue(memtable.getAllocator().offHeap().owns() > 0);
-    }
-
     @Test
     public void randomQueryTest() throws Exception
     {
         memtableIndex = new TrieMemtableIndex(indexContext, memtable);
         assertEquals(TrieMemtable.SHARD_COUNT, memtableIndex.shardCount());
+        assertEquals(TrieMemtable.BUFFER_TYPE, BufferType.OFF_HEAP);
```

Comment:
Does the test fail for ON_HEAP?

Also, this suggests that we don't have a test configuration that will exercise ON_HEAP allocation for these classes; I think that's valuable so I would move all of these methods to the base class and let them run for both.

Reply:
It was never written to run with ON_HEAP, but I will add coverage for all tests with ON_HEAP too as agreed, as improvement